astropy/io/fits/column.py

```diff
@@ -2470,9 +2474,13 @@ def _convert_ascii_format(format, reverse=False):
 
         # The following logic is taken from CFITSIO:
         # For integers, if the width <= 4 we can safely use 16-bit ints for all
+        # values, if width >= 10 we may need to accomodate 64-bit ints.
         # values [for the non-standard J format code just always force 64-bit]
-        if format == 'I' and width <= 4:
-            recformat = 'i2'
+        if format == 'I':
+            if width <= 4:
+                recformat = 'i2'
+            elif width > 9:
+                recformat = 'i8'
```

Comment:
I wonder if there should be a warning in the case of widths beyond that supported by a 64-bit integer.  Of course the problem with this format is you can have a column with a format like `I50` but that contains mostly whitespace in each entry, while the actual values are much smaller.